File: JavaScript/condicionales.js
const Pi = 3.1416;
const constValue = 2;

const readlineSync = require('readline-sync'); // Para leer la entrada del usuario//

// Solicitar la figura a calcular//
console.log("Seleccione la figura a calcular su área:");
console.log("1 para rombo");
console.log("2 para rectángulo");
console.log("3 para cuadrado");
console.log("4 para trapecio");

let figura = readlineSync.question('Elige una opción (1, 2, 3, 4): ');

switch (figura) {
    case '1':
        let Dmayor = parseInt(readlineSync.question("Ingresa el valor de la diagonal mayor: "));
        let Dmenor = parseInt(readlineSync.question("Ingresa el valor de la diagonal menor: "));
        let areaRombo = (Dmayor * Dmenor) / constValue;
        console.log(`El área del rombo es: ${areaRombo}`);
        break;

    case '2':
        let largo = parseInt(readlineSync.question("Ingresa el valor del largo del rectángulo: "));
        let ancho = parseInt(readlineSync.question("Ingresa el valor del ancho del rectángulo: "));
        let areaRectangulo = largo * ancho;
        console.log(`El área del rectángulo es: ${areaRectangulo}`);
        break;

    case '3':
        let lado = parseInt(readlineSync.question("Ingresa el valor del lado del cuadrado: "));
        let areaCuadrado = lado * lado;
        console.log(`El área del cuadrado es: ${areaCuadrado}`);
        break;

    case '4':
        let Bmayor = parseInt(readlineSync.question("Ingresa el valor de la base mayor: "));
        let Bmenor = parseInt(readlineSync.question("Ingresa el valor de la base menor: "));
        let altura = parseInt(readlineSync.question("Ingresa la altura del trapecio: "));
        let areaTrapecio = ((Bmayor + Bmenor) * altura) / 2;
        console.log(`El área del trapecio es: ${areaTrapecio}`);
        break;

    default:
        console.log("Opción no válida.");
        break;
}

File: go/condicionales.go
package main

import (
	"fmt"
)

func main() {
	var Figura string
	const Pi = 3.1416
	const constante = 2

	// Selección de la figura
	fmt.Print("Seleccione la figura a calcular su área:\n\n 1 para rombo\n\n 2 para rectángulo\n\n 3 para cuadrado\n\n 4 para trapecio\n\n: ")
	fmt.Scan(&Figura)

	// Condiciones para calcular el área dependiendo de la figura seleccionada
	switch Figura {
	case "1":
		// Rombo
		var Dmayor, Dmenor int
		fmt.Print("Ingresa el valor de la diagonal mayor: ")
		fmt.Scan(&Dmayor)
		fmt.Print("Ingresa el valor de la diagonal menor: ")
		fmt.Scan(&Dmenor)
		Area := float64(Dmayor+Dmenor) / float64(constante)
		fmt.Printf("El área del rombo es: %.2f\n", Area)

	case "2":
		// Rectángulo
		var Largo, Ancho int
		fmt.Print("Ingresa el valor del largo del rectángulo: ")
		fmt.Scan(&Largo)
		fmt.Print("Ingresa el valor del ancho del rectángulo: ")
		fmt.Scan(&Ancho)
		Area := float64(Largo * Ancho)
		fmt.Printf("El área del rectángulo es: %.2f\n", Area)

	case "3":
		// Cuadrado
		var Lado int
		fmt.Print("Ingresa el valor del lado del cuadrado: ")
		fmt.Scan(&Lado)
		Area := float64(Lado * Lado)
		fmt.Printf("El área del cuadrado es: %.2f\n", Area)

	case "4":
		// Trapecio
		var Bmayor, Bmenor, H int
		fmt.Print("Ingresa el valor de la base mayor: ")
		fmt.Scan(&Bmayor)
		fmt.Print("Ingresa el valor de la base menor: ")
		fmt.Scan(&Bmenor)
		fmt.Print("Ingresa la altura del trapecio: ")
		fmt.Scan(&H)
		Area := float64((Bmayor+Bmenor)*H) / 2
		fmt.Printf("El área del trapecio es: %.2f\n", Area)

	default:
		// En caso de no seleccionar ninguna opción válida, calculamos un área circular
		var A float64
		A = Pi
		fmt.Printf("El área del círculo es: %.2f\n", A)
	}
}
